ruby_enumerable_methods.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    select_arr = []
+    for item in self do
+      select_arr << item if yield(item)
+    end
+    select_arr
+  end
+
+  def my_all?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break unless pass
+    end
+    pass
+  end
+
+  def my_any?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break if pass
+    end
+    pass
+  end
```

Comment:
- enumerables my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
     Failure/Error: expect(true_array.my_any?).to be true_array.any?

- enumerables my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
     Failure/Error: expect(array.my_any?(Integer)).to be array.any?(Integer)

-  enumerables my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
      Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)

- enumerables my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')